salt/modules/boto3_route53.py

```diff
@@ -714,6 +714,74 @@ def delete_hosted_zone_by_domain(Name, PrivateZone=None, region=None, key=None,
 
+def _to_aws_encoding(instring):
+    '''
+    A partial implememtation of the `AWS domain encoding`__ rules. The punycode section is
+    ignored for now, so you'll have to input any unicode characters already punycoded.
+
+    .. __: http://docs.aws.amazon.com/Route53/latest/DeveloperGuide/DomainNameFormat.html
+
+    The only breakage is that we don't permit ASCII 92 ('backslash') in domain names, even though
+    AWS seems to allow it, becuase it's used to introduce octal escapes, and I can't think of a
+    good way to allow it while still parsing those escapes.  Imagine a domain name containing the
+    literal string '\134' - pathological to be sure, but allowed by AWS's rules.  If someone
+    dislikes this restriction, feel free to update this function to correctly handle such madness.
+
+    Oh, and the idea of allowing '.' (period) within a domain component (e.g. not as a separator,
+    but as part of a field) is as stupid it sounds, and is also not supported by these functions.
+    Again, if you can figure out a sensible way to make it work, more power too you.  If you
+    REALLY need this, just embed '\056' directly in your strings.  I can't promise it won't break
+    your DNS out in the real world though.
+    '''
+    instring = instring.lower()                                           # Always lowercase
+    send_as_is = list(range(97, 123))                   # a-z is 97-122 inclusive
+    send_as_is += [ord(n) for n in ['0', '1', '2', '3', '4', '5', '6', '7', '8',
+                                    '9', '-', '_', '.']]
+    send_as_octal = [ord(n) for n in ['!', '"', '#', '$', '%', '&', "'", '(', ')', '*',
+                                      '+', ',', '/', '\\', ':', ';', '<', '=', '>', '?',
+                                      '@', '[', ']', '^', '`', '{', '|', '}']]
+    # Non-printable ASCII - AWS claims it's valid in DNS entries... YMMV
+    acceptable_points_which_must_be_octalized = list(range(0, 33))        # 0-32 inclusive
+    # "Extended ASCII" stuff
+    acceptable_points_which_must_be_octalized += list(range(127, 256))    # 127-255 inclusive
+    # ^^^ This, BTW, is incompatible with punycode as far as I can tell, since unicode only aligns
+    # with ASCII for the first 127 code-points...  Oh well.
+    outlist = []
+    for char in instring:
+        point = ord(char)
+        octal = '\\{:03o}'.format(point)
+        if point in send_as_is:
+            outlist += [char]
+        elif point in send_as_octal:
+            outlist += [octal]
+        elif point in acceptable_points_which_must_be_octalized:
+            outlist += [octal]
+        else:
+            raise SaltInvocationError("Invalid Route53 domain character seen (octal {0}) in string "
+                                      "{1}.  Do you need to punycode it?".format(octal, instring))
+    ret = ''.join(outlist)
+    log.debug('Name after massaging is: {}'.format(ret))
```

Comment:
Please use % formatting here. Also, keep in mind that log messages should be distinctive. "Name after massaging" won't necessarily mean anything to someone looking at debug logging. My suggestion is to use something like:

```python
log.debug('AWS-encoded result for %s: %s', instring, ret)
```